tinyiiod.c

```diff
@@ -41,19 +45,27 @@ struct tinyiiod * tinyiiod_create(const char *xml,
 	return iiod;
 }
 
+/***************************************************************************//**
+ * @brief tinyiiod_destroy
+*******************************************************************************/
 void tinyiiod_destroy(struct tinyiiod *iiod)
 {
 	free(iiod);
 }
 
-int tinyiiod_read_command(struct tinyiiod *iiod)
+/***************************************************************************//**
+ * @brief tinyiiod_read_command
+*******************************************************************************/
+int32_t tinyiiod_read_command(struct tinyiiod *iiod)
 {
-	char buf[128];
-	int ret;
+	char buf[128] = {0};
```

Comment:
initializing a 128 buffer size on the stack, is a bit costly for some micro-controllers;
it may be non-standard, since this construct may only be accepted by some compilers and not by others;
it's also un-needed; for buffers, you typically keep an offset & length of data where you look into/for